docs/content/latest/api/ycql/explain.md

```diff
@@ -0,0 +1,211 @@
+   ... ) WITH transactions = { 'enabled' : true };
+cqlsh> CREATE INDEX IF NOT EXISTS most_watched_by_year
+   ...   ON imdb.movie_stats((movie_genre, last_watched), movie_name, user_name)
+   ...   INCLUDE(user_rank);
+cqlsh> CREATE INDEX IF NOT EXISTS best_rated
+   ...   ON imdb.movie_stats((user_rank, movie_genre), movie_name, user_name)
+   ...   INCLUDE(last_watched);
+```
+
+Insert some rows.
+```CQL
+cqlsh> USE imdb;
+cqlsh:imdb> INSERT INTO movie_stats(movie_name, movie_genre, user_name, user_rank, last_watched)
+        ...     VALUES ('m1', 'g1', 'u1', 5, '2019-01-18');
+cqlsh:imdb> INSERT INTO movie_stats(movie_name, movie_genre, user_name, user_rank, last_watched)
+        ...     VALUES ('m2', 'g2', 'u1', 4, '2019-01-17');
+cqlsh:imdb> INSERT INTO movie_stats(movie_name, movie_genre, user_name, user_rank, last_watched)
+        ...     VALUES ('m3', 'g1', 'u2', 5, '2019-01-18');
+cqlsh:imdb> INSERT INTO movie_stats(movie_name, movie_genre, user_name, user_rank, last_watched)
+        ...     VALUES ('m4', 'g1', 'u1', 2, '2019-02-27');
+```
+If movie_genre, or movie_genre & movie_name, or movie_genre & movie_name & user_name are specified, the query should be served efficiently from the primary table.
+
+```CQL
+cqlsh:imdb> EXPLAIN SELECT *
+        ...   FROM movie_stats
+        ...  WHERE movie_genre = 'g1';
+
+QUERY PLAN
+----------------------------------------
+ Range Scan on imdb.movie_stats
+   Key Conditions: (movie_genre = 'g1')
+```
+If movie_genre & last_watched are specified, then the query should be served efficiently from the `most_watched_by_year` index.
+
+```CQL
+cqlsh:imdb> EXPLAIN SELECT *
+        ...   FROM movie_stats
+        ...  WHERE movie_genre = 'g1' and last_watched='2019-02-27';
+
+QUERY PLAN
+--------------------------------------------------------------------------
+ Index Only Scan using imdb.most_watched_by_year on imdb.movie_stats
+   Key Conditions: (movie_genre = 'g1') AND (last_watched = '2019-02-27')
+
+```
+
+If user_rank and movie_genre are specified then the query should be served efficiently from the `best_rated` index.
+
+```CQL
+cqlsh:imdb> EXPLAIN SELECT *
+        ...   FROM movie_stats
+        ...  WHERE movie_genre = 'g2' and user_rank=5;
+
+QUERY PLAN
+--------------------------------------------------------------
+ Index Only Scan using imdb.best_rated on imdb.movie_stats
+   Key Conditions: (user_rank = '5') AND (movie_genre = 'g2')
+
+```
+Create non-covering index.
+```CQL
+cqlsh:imdb> DROP INDEX best_rated;
+cqlsh:imdb> CREATE INDEX IF NOT EXISTS best_rated
+        ...   ON imdb.movie_stats((user_rank, movie_genre), movie_name, user_name);
+```
+2-Step select. Using Index Scan as opposed to Index Only Scan.
+```CQL
+cqlsh:imdb> EXPLAIN SELECT *
+        ...   FROM movie_stats
+        ...  WHERE movie_genre = 'g2' and user_rank=5;
+
+ QUERY PLAN
+--------------------------------------------------------------
+ Index Scan using imdb.best_rated on imdb.movie_stats
```

Comment:
We should also add a note explaining the difference between `Index Scan` and `Index only Scan`

Reply:
See https://github.com/yugabyte/yugabyte-db/pull/4174/commits/cc9562b5cf66ebdae4705b4679c230e3f6dbbe4e

Reply:
Added in commit above